### PR TITLE
Add configuration options for AWS S3 server access logging

### LINKED ISSUE #3003: Expose additional configuration options for AWS S3 server access logging
**Describe the solution you'd like**

In our environment, the AWS S3 bucket where S3 access logs should be stored is pre-created by the "central" team, and we don't have permissions to make any changes to it. Therefore, we don't need to:
- Create it: The current code creates the logging bucket only if it doesn't exist, which is already appropriate.
- Change configuration of the logging bucket, such as ACL, public access, encryption, or TLS enforcement.

Currently, we cannot configure [server access logging](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ServerLogs.html) because we have to use the pre-created bucket. Additionally, it would be beneficial to have the option to configure `PartitionDateSource` in the logging configuration.
Example:

```
{
  "LoggingEnabled": {
    "TargetBucket": "<bucket_name> ",
    "TargetPrefix": "",
    "TargetObjectKeyFormat": {
      "PartitionedPrefix": {
        "PartitionDateSource": "EventTime"
      }
    }
  }
}
```

There is also a **small bug** because the current code silently fails on the permission error when attempting to configure the logging bucket.

**Describe alternatives you've considered**

The only alternative is to configure access logging outside of Terragrunt.

### PR DESCRIPTION
**Pull request by findmyname666**
## Description

Fixes #3003.

In our environment the S3 bucket where should be stored S3 access logs is pre-created by "central" team and we don't have permissions to do any changes. In such case we don't have to:
- Create it - the current code creates the logging bucket only if it doesn't exist. No change is needed.
- Configure logging bucket e.g. ACL, public access, encryption, tls enforcement - adding option to skip the configuration for access log bucket.
- PartitionDateSource - Adding one more option to allow users to configure `PartitionDateSource`.

Additionally 1) adding error logging into the function `configureAccessLogBucket`. Without it I wasn't able to detect why the logging isn't configured on the source bucket. The error is returned but ignored. 2) Adding code to detect if the logging policy / configuration was change. The previous code just check if some policy is enabled.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

### Migration Guide

N/A.

### CODE REVIEW

File: remote/remote_state_s3.go [hunk 1]
```diff
@@ -39,23 +39,28 @@ const (
...
+	DisableBucketUpdate                          bool              `mapstructure:"disable_bucket_update"`
+	EnableLockTableSSEncryption                  bool              `mapstructure:"enable_lock_table_ssencryption"`
+	DisableAWSClientChecksums                    bool              `mapstructure:"disable_aws_client_checksums"`
+	AccessLoggingBucketName                      string            `mapstructure:"accesslogging_bucket_name"`
+	AccessLoggingTargetObjectPartitionDateSource string            `mapstructure:"accesslogging_target_object_partition_date_source"`
+	AccessLoggingTargetPrefix                    string            `mapstructure:"accesslogging_target_prefix"`
+	SkipAccessLoggingBucketAcl                   bool              `mapstructure:"skip_accesslogging_bucket_acl"`
+	SkipAccessLoggingBucketEnforcedTLS           bool              `mapstructure:"skip_accesslogging_bucket_enforced_tls"`
+	SkipAccessLoggingBucketPublicAccessBlocking  bool              `mapstructure:"skip_accesslogging_bucket_public_access_blocking"`
+	SkipAccessLoggingBucketSSEncryption          bool              `mapstructure:"skip_accesslogging_bucket_ssencryption"`
```
[the remark this answers is not in the document]
Comment (findmyname666) [reply remote/remote_state_s3.go:61]: Hi @denis256, Thx for your feedback. I tried to updated existing existing tests and added one new for func `createS3LoggingInput`. Please let me know if I missed anything else.

[the remark this answers is not in the document]
Comment (findmyname666) [reply remote/remote_state_s3.go:61]: Hi @denis256 , Can I politely ask you for re-review once you have some free time please? Thx.

[the remark this answers is not in the document]
Comment (findmyname666) [reply remote/remote_state_s3.go:61]: Hi @denis256, do you think you'll have time to take a look on this PR in the upcoming days?

**Comment by yhakbar** [comment]
Hey @findmyname666 , are you willing to rebase to get this PR up to speed?

Also, are you able to run the tests locally to confirm that they work as expected?

**Comment by findmyname666** [comment]
> Hey @findmyname666 , are you willing to rebase to get this PR up to speed?
> 
> Also, are you able to run the tests locally to confirm that they work as expected?

Hi @yhakbar , I will try to do it this week. Thx.

**Comment by findmyname666** [comment]
> Hey @findmyname666 , are you willing to rebase to get this PR up to speed?
> 
> Also, are you able to run the tests locally to confirm that they work as expected?

Hi @yhakbar,

I re-based the code and re-tested `remote` package:
```
$ go test -v
=== RUN   TestGCSConfigValuesEqual
=== PAUSE TestGCSConfigValuesEqual
=== RUN   TestCreateS3LoggingInput
=== PAUSE TestCreateS3LoggingInput
=== RUN   TestConfigValuesEqual
=== PAUSE TestConfigValuesEqual
=== RUN   TestForcePathStyleClientSession
=== PAUSE TestForcePathStyleClientSession
=== RUN   TestGetAwsSessionConfig
=== PAUSE TestGetAwsSessionConfig
=== RUN   TestGetAwsSessionConfigWithAssumeRole
=== PAUSE TestGetAwsSessionConfigWithAssumeRole
=== RUN   TestGetTerraformInitArgs
=== PAUSE TestGetTerraformInitArgs
=== RUN   TestNegativePublicAccessResponse
=== PAUSE TestNegativePublicAccessResponse
=== RUN   TestValidateS3Config
=== PAUSE TestValidateS3Config
=== RUN   TestToTerraformInitArgs
=== PAUSE TestToTerraformInitArgs
=== RUN   TestToTerraformInitArgsForGCS
=== PAUSE TestToTerraformInitArgsForGCS
=== RUN   TestToTerraformInitArgsUnknownBackend
=== PAUSE TestToTerraformInitArgsUnknownBackend
=== RUN   TestToTerraformInitArgsInitDisabled
=== PAUSE TestToTerraformInitArgsInitDisabled
=== RUN   TestToTerraformInitArgsNoBackendConfigs
=== PAUSE TestToTerraformInitArgsNoBackendConfigs
=== RUN   TestDiffersFrom
=== PAUSE TestDiffersFrom
=== RUN   TestWrapMapToSingleLineHcl
=== RUN   TestWrapMapToSingleLineHcl/SimpleMap
=== RUN   TestWrapMapToSingleLineHcl/NestedMap
--- PASS: TestWrapMapToSingleLineHcl (0.00s)
    --- PASS: TestWrapMapToSingleLineHcl/SimpleMap (0.00s)
    --- PASS: TestWrapMapToSingleLineHcl/NestedMap (0.00s)
=== RUN   TestParseTerraformStateLocal
=== PAUSE TestParseTerraformStateLocal
=== RUN   TestParseTerraformStateRemote
=== PAUSE TestParseTerraformStateRemote
=== RUN   TestParseTerraformStateRemoteFull
=== PAUSE TestParseTerraformStateRemoteFull
=== RUN   TestParseTerraformStateEmpty
=== PAUSE TestParseTerraformStateEmpty
=== RUN   TestParseTerraformStateInvalid
=== PAUSE TestParseTerraformStateInvalid
=== CONT  TestGCSConfigValuesEqual
=== CONT  TestDiffersFrom
=== RUN   TestGCSConfigValuesEqual/equal-both-empty
=== PAUSE TestGCSConfigValuesEqual/equal-both-empty
=== RUN   TestGCSConfigValuesEqual/equal-empty-and-nil
=== RUN   TestDiffersFrom/both_empty
=== PAUSE TestGCSConfigValuesEqual/equal-empty-and-nil
=== PAUSE TestDiffersFrom/both_empty
=== RUN   TestGCSConfigValuesEqual/equal-empty-and-nil-backend-config
=== RUN   TestDiffersFrom/same_backend_type_value
=== PAUSE TestDiffersFrom/same_backend_type_value
=== RUN   TestDiffersFrom/different_backend_type_values
=== PAUSE TestDiffersFrom/different_backend_type_values
=== RUN   TestDiffersFrom/identical_S3_configs
=== PAUSE TestDiffersFrom/identical_S3_configs
=== RUN   TestDiffersFrom/identical_GCS_configs
=== PAUSE TestDiffersFrom/identical_GCS_configs
=== CONT  TestParseTerraformStateRemoteFull
=== PAUSE TestGCSConfigValuesEqual/equal-empty-and-nil-backend-config
=== CONT  TestParseTerraformStateRemote
=== RUN   TestGCSConfigValuesEqual/equal-one-key
=== PAUSE TestGCSConfigValuesEqual/equal-one-key
=== CONT  TestNegativePublicAccessResponse
--- PASS: TestParseTerraformStateRemote (0.00s)
=== CONT  TestToTerraformInitArgsInitDisabled
--- PASS: TestToTerraformInitArgsInitDisabled (0.00s)
=== CONT  TestToTerraformInitArgsUnknownBackend
=== RUN   TestNegativePublicAccessResponse/nil-response
--- PASS: TestToTerraformInitArgsUnknownBackend (0.00s)
=== CONT  TestToTerraformInitArgsForGCS
=== PAUSE TestNegativePublicAccessResponse/nil-response
=== CONT  TestToTerraformInitArgs
=== RUN   TestNegativePublicAccessResponse/legacy-bucket
=== PAUSE TestNegativePublicAccessResponse/legacy-bucket
=== CONT  TestToTerraformInitArgsNoBackendConfigs
=== RUN   TestNegativePublicAccessResponse/false-response
=== CONT  TestValidateS3Config
=== PAUSE TestNegativePublicAccessResponse/false-response
=== CONT  TestGetAwsSessionConfig
=== CONT  TestGetAwsSessionConfigWithAssumeRole
=== RUN   TestGetAwsSessionConfigWithAssumeRole/all-values
=== PAUSE TestGetAwsSessionConfigWithAssumeRole/all-values
--- PASS: TestToTerraformInitArgsForGCS (0.00s)
=== CONT  TestConfigValuesEqual
=== RUN   TestGetAwsSessionConfig/all-values
=== RUN   TestValidateS3Config/no-region
=== PAUSE TestGetAwsSessionConfig/all-values
=== RUN   TestGetAwsSessionConfig/no-values
=== CONT  TestParseTerraformStateEmpty
=== PAUSE TestGetAwsSessionConfig/no-values
=== CONT  TestForcePathStyleClientSession
=== RUN   TestGetAwsSessionConfig/extra-values
=== PAUSE TestGetAwsSessionConfig/extra-values
=== RUN   TestDiffersFrom/different_s3_bucket_values
=== PAUSE TestDiffersFrom/different_s3_bucket_values
=== RUN   TestDiffersFrom/different_gcs_bucket_values
=== PAUSE TestDiffersFrom/different_gcs_bucket_values
=== CONT  TestCreateS3LoggingInput
=== RUN   TestGCSConfigValuesEqual/equal-multiple-keys
=== PAUSE TestGCSConfigValuesEqual/equal-multiple-keys
=== RUN   TestCreateS3LoggingInput/equal-default-prefix-no-partition-date-source
=== PAUSE TestCreateS3LoggingInput/equal-default-prefix-no-partition-date-source
=== RUN   TestForcePathStyleClientSession/path-style-true
=== RUN   TestCreateS3LoggingInput/equal-no-prefix-no-partition-date-source
=== PAUSE TestForcePathStyleClientSession/path-style-true
=== PAUSE TestCreateS3LoggingInput/equal-no-prefix-no-partition-date-source
=== RUN   TestCreateS3LoggingInput/equal-custom-prefix-no-partition-date-source
=== PAUSE TestCreateS3LoggingInput/equal-custom-prefix-no-partition-date-source
=== RUN   TestCreateS3LoggingInput/equal-custom-prefix-custom-partition-date-source
=== RUN   TestDiffersFrom/different_s3_key_values
=== PAUSE TestCreateS3LoggingInput/equal-custom-prefix-custom-partition-date-source
=== PAUSE TestDiffersFrom/different_s3_key_values
=== RUN   TestGCSConfigValuesEqual/equal-encrypt-bool-handling
=== PAUSE TestGCSConfigValuesEqual/equal-encrypt-bool-handling
=== RUN   TestDiffersFrom/different_gcs_prefix_values
=== RUN   TestGCSConfigValuesEqual/equal-general-bool-handling
=== PAUSE TestDiffersFrom/different_gcs_prefix_values
=== PAUSE TestGCSConfigValuesEqual/equal-general-bool-handling
=== RUN   TestDiffersFrom/different_s3_region_values
=== RUN   TestGCSConfigValuesEqual/equal-ignore-gcs-labels
=== PAUSE TestDiffersFrom/different_s3_region_values
=== RUN   TestDiffersFrom/different_gcs_location_values
--- PASS: TestToTerraformInitArgs (0.00s)
=== PAUSE TestDiffersFrom/different_gcs_location_values
--- PASS: TestToTerraformInitArgsNoBackendConfigs (0.00s)
=== RUN   TestDiffersFrom/different_boolean_values_and_boolean_conversion
=== PAUSE TestDiffersFrom/different_boolean_values_and_boolean_conversion
=== RUN   TestDiffersFrom/different_gcs_boolean_values_and_boolean_conversion
=== PAUSE TestDiffersFrom/different_gcs_boolean_values_and_boolean_conversion
=== RUN   TestDiffersFrom/null_values_ignored
=== PAUSE TestDiffersFrom/null_values_ignored
=== RUN   TestDiffersFrom/gcs_null_values_ignored
=== PAUSE TestDiffersFrom/gcs_null_values_ignored
=== PAUSE TestGCSConfigValuesEqual/equal-ignore-gcs-labels
=== CONT  TestNegativePublicAccessResponse/nil-response
--- PASS: TestParseTerraformStateRemoteFull (0.00s)
=== RUN   TestGCSConfigValuesEqual/unequal-wrong-backend
=== CONT  TestNegativePublicAccessResponse/false-response
=== PAUSE TestGCSConfigValuesEqual/unequal-wrong-backend
=== CONT  TestParseTerraformStateLocal
=== CONT  TestNegativePublicAccessResponse/legacy-bucket
=== CONT  TestGetTerraformInitArgs
=== RUN   TestGCSConfigValuesEqual/unequal-values
=== PAUSE TestGCSConfigValuesEqual/unequal-values
=== RUN   TestGetTerraformInitArgs/empty-no-values
=== RUN   TestGCSConfigValuesEqual/unequal-non-empty-config-nil
=== CONT  TestGetAwsSessionConfig/all-values
=== CONT  TestGetAwsSessionConfigWithAssumeRole/all-values
=== CONT  TestGetAwsSessionConfig/no-values
=== CONT  TestGetAwsSessionConfig/extra-values
=== CONT  TestCreateS3LoggingInput/equal-default-prefix-no-partition-date-source
=== CONT  TestCreateS3LoggingInput/equal-no-prefix-no-partition-date-source
=== CONT  TestCreateS3LoggingInput/equal-custom-prefix-no-partition-date-source
=== CONT  TestCreateS3LoggingInput/equal-custom-prefix-custom-partition-date-source
=== RUN   TestConfigValuesEqual/equal-both-empty
=== PAUSE TestConfigValuesEqual/equal-both-empty
=== RUN   TestForcePathStyleClientSession/path-style-false
=== RUN   TestValidateS3Config/no-bucket
=== PAUSE TestForcePathStyleClientSession/path-style-false
--- PASS: TestParseTerraformStateEmpty (0.00s)
=== CONT  TestDiffersFrom/both_empty
=== RUN   TestForcePathStyleClientSession/path-style-non-existent
--- PASS: TestNegativePublicAccessResponse (0.00s)
    --- PASS: TestNegativePublicAccessResponse/nil-response (0.00s)
    --- PASS: TestNegativePublicAccessResponse/false-response (0.00s)
    --- PASS: TestNegativePublicAccessResponse/legacy-bucket (0.00s)
=== PAUSE TestForcePathStyleClientSession/path-style-non-existent
=== CONT  TestParseTerraformStateInvalid
=== RUN   TestValidateS3Config/no-key
=== CONT  TestDiffersFrom/different_gcs_bucket_values
DEBU[0000] Backend  has not changed.                    
=== PAUSE TestGCSConfigValuesEqual/unequal-non-empty-config-nil
=== CONT  TestDiffersFrom/different_s3_bucket_values
=== RUN   TestGCSConfigValuesEqual/unequal-general-bool-handling
=== RUN   TestConfigValuesEqual/equal-empty-and-nil
=== PAUSE TestGCSConfigValuesEqual/unequal-general-bool-handling
=== PAUSE TestConfigValuesEqual/equal-empty-and-nil
=== RUN   TestGCSConfigValuesEqual/equal-null-ignored
DEBU[0000] Changed from map[bucket:foo key:bar region:us-east-1] to map[bucket:different key:bar region:us-east-1] 
=== PAUSE TestGCSConfigValuesEqual/equal-null-ignored
=== RUN   TestConfigValuesEqual/equal-empty-and-nil-backend-config
=== CONT  TestDiffersFrom/identical_GCS_configs
=== PAUSE TestConfigValuesEqual/equal-empty-and-nil-backend-config
=== RUN   TestGCSConfigValuesEqual/terragrunt-only-configs-remain-intact
DEBU[0000] Changed from map[bucket:foo location:europe-west3 prefix:bar project:foo-123456] to map[bucket:different location:europe-west3 prefix:bar project:foo-123456] 
=== RUN   TestConfigValuesEqual/equal-one-key
=== PAUSE TestConfigValuesEqual/equal-one-key
=== PAUSE TestGCSConfigValuesEqual/terragrunt-only-configs-remain-intact
DEBU[0000] Backend gcs has not changed.                 
--- PASS: TestParseTerraformStateLocal (0.00s)
=== CONT  TestDiffersFrom/same_backend_type_value
=== CONT  TestDiffersFrom/different_gcs_prefix_values
=== CONT  TestDiffersFrom/different_s3_key_values
--- PASS: TestGetAwsSessionConfigWithAssumeRole (0.00s)
    --- PASS: TestGetAwsSessionConfigWithAssumeRole/all-values (0.00s)
=== PAUSE TestGetTerraformInitArgs/empty-no-values
DEBU[0000] Backend s3 has not changed.                  
=== CONT  TestDiffersFrom/identical_S3_configs
=== CONT  TestDiffersFrom/gcs_null_values_ignored
=== RUN   TestGetTerraformInitArgs/valid-s3-configuration-keys
DEBU[0000] Backend s3 has not changed.                  
=== RUN   TestConfigValuesEqual/equal-multiple-keys
=== PAUSE TestGetTerraformInitArgs/valid-s3-configuration-keys
=== PAUSE TestConfigValuesEqual/equal-multiple-keys
DEBU[0000] Backend gcs has not changed.                 
=== RUN   TestGetTerraformInitArgs/terragrunt-keys-filtered
=== RUN   TestConfigValuesEqual/equal-encrypt-bool-handling
=== PAUSE TestGetTerraformInitArgs/terragrunt-keys-filtered
DEBU[0000] Changed from map[bucket:foo location:europe-west3 prefix:bar project:foo-123456] to map[bucket:foo location:europe-west3 prefix:different project:foo-123456] 
=== PAUSE TestConfigValuesEqual/equal-encrypt-bool-handling
=== CONT  TestDiffersFrom/different_boolean_values_and_boolean_conversion
DEBU[0000] Changed from map[bucket:foo key:bar region:us-east-1] to map[bucket:foo key:different region:us-east-1] 
=== RUN   TestConfigValuesEqual/equal-general-bool-handling
=== PAUSE TestConfigValuesEqual/equal-general-bool-handling
=== CONT  TestDiffersFrom/different_gcs_boolean_values_and_boolean_conversion
DEBU[0000] Changed from map[something:true] to map[something:%!s(bool=false)] 
=== RUN   TestConfigValuesEqual/equal-ignore-s3-tags
DEBU[0000] Changed from map[something:true] to map[something:%!s(bool=false)] 
=== PAUSE TestConfigValuesEqual/equal-ignore-s3-tags
=== CONT  TestDiffersFrom/different_s3_region_values
=== RUN   TestConfigValuesEqual/equal-ignore-dynamodb-tags
=== PAUSE TestConfigValuesEqual/equal-ignore-dynamodb-tags
=== RUN   TestConfigValuesEqual/equal-ignore-accesslogging-options
=== PAUSE TestConfigValuesEqual/equal-ignore-accesslogging-options
DEBU[0000] Changed from map[bucket:foo key:bar region:us-east-1] to map[bucket:foo key:bar region:different] 
=== RUN   TestConfigValuesEqual/unequal-wrong-backend
=== CONT  TestForcePathStyleClientSession/path-style-non-existent
=== CONT  TestForcePathStyleClientSession/path-style-true
=== CONT  TestDiffersFrom/different_backend_type_values
=== PAUSE TestConfigValuesEqual/unequal-wrong-backend
INFO[0000] Backend type has changed from s3 to atlas    
=== RUN   TestGetTerraformInitArgs/empty-no-values-all-terragrunt-keys-filtered
=== RUN   TestConfigValuesEqual/unequal-values
=== CONT  TestForcePathStyleClientSession/path-style-false
=== PAUSE TestGetTerraformInitArgs/empty-no-values-all-terragrunt-keys-filtered
=== PAUSE TestConfigValuesEqual/unequal-values
=== RUN   TestGetTerraformInitArgs/lock-table-replaced-with-dynamodb-table
=== PAUSE TestGetTerraformInitArgs/lock-table-replaced-with-dynamodb-table
=== RUN   TestGetTerraformInitArgs/dynamodb-table-not-replaced-with-lock-table
=== PAUSE TestGetTerraformInitArgs/dynamodb-table-not-replaced-with-lock-table
=== RUN   TestGetTerraformInitArgs/assume-role
=== PAUSE TestGetTerraformInitArgs/assume-role
=== RUN   TestConfigValuesEqual/unequal-non-empty-config-nil
=== PAUSE TestConfigValuesEqual/unequal-non-empty-config-nil
=== CONT  TestGCSConfigValuesEqual/equal-both-empty
=== CONT  TestDiffersFrom/null_values_ignored
=== CONT  TestGCSConfigValuesEqual/unequal-wrong-backend
DEBU[0000] Backend s3 has not changed.                  
DEBU[0000] Backend type has changed from gcs to wrong   
=== CONT  TestGCSConfigValuesEqual/equal-ignore-gcs-labels
=== RUN   TestConfigValuesEqual/unequal-general-bool-handling
=== CONT  TestGCSConfigValuesEqual/equal-general-bool-handling
=== CONT  TestGCSConfigValuesEqual/equal-encrypt-bool-handling
=== PAUSE TestConfigValuesEqual/unequal-general-bool-handling
=== CONT  TestGCSConfigValuesEqual/equal-multiple-keys
=== CONT  TestGCSConfigValuesEqual/equal-one-key
=== RUN   TestConfigValuesEqual/equal-null-ignored
=== PAUSE TestConfigValuesEqual/equal-null-ignored
=== CONT  TestGCSConfigValuesEqual/unequal-non-empty-config-nil
=== CONT  TestGCSConfigValuesEqual/unequal-general-bool-handling
=== CONT  TestGCSConfigValuesEqual/equal-empty-and-nil
=== CONT  TestGCSConfigValuesEqual/terragrunt-only-configs-remain-intact
=== CONT  TestGCSConfigValuesEqual/equal-null-ignored
=== CONT  TestGetTerraformInitArgs/empty-no-values
--- PASS: TestGetAwsSessionConfig (0.00s)
    --- PASS: TestGetAwsSessionConfig/all-values (0.00s)
    --- PASS: TestGetAwsSessionConfig/no-values (0.00s)
    --- PASS: TestGetAwsSessionConfig/extra-values (0.00s)
DEBU[0000] Backend config changed from map[something:%!s(bool=false)] to map[something:%!s(bool=true)] 
=== CONT  TestGCSConfigValuesEqual/unequal-values
=== CONT  TestGetTerraformInitArgs/empty-no-values-all-terragrunt-keys-filtered
=== CONT  TestGetTerraformInitArgs/terragrunt-keys-filtered
=== CONT  TestGetTerraformInitArgs/valid-s3-configuration-keys
=== CONT  TestGetTerraformInitArgs/lock-table-replaced-with-dynamodb-table
=== CONT  TestGetTerraformInitArgs/dynamodb-table-not-replaced-with-lock-table
--- PASS: TestCreateS3LoggingInput (0.00s)
    --- PASS: TestCreateS3LoggingInput/equal-default-prefix-no-partition-date-source (0.00s)
    --- PASS: TestCreateS3LoggingInput/equal-no-prefix-no-partition-date-source (0.00s)
    --- PASS: TestCreateS3LoggingInput/equal-custom-prefix-no-partition-date-source (0.00s)
    --- PASS: TestCreateS3LoggingInput/equal-custom-prefix-custom-partition-date-source (0.00s)
DEBU[0000] Backend config changed from map[foo:different] to map[foo:bar] 
=== CONT  TestDiffersFrom/different_gcs_location_values
=== CONT  TestGCSConfigValuesEqual/equal-empty-and-nil-backend-config
=== CONT  TestConfigValuesEqual/equal-null-ignored
=== CONT  TestConfigValuesEqual/equal-ignore-accesslogging-options
=== CONT  TestConfigValuesEqual/equal-ignore-dynamodb-tags
=== CONT  TestConfigValuesEqual/equal-both-empty
=== CONT  TestConfigValuesEqual/equal-general-bool-handling
=== CONT  TestGetTerraformInitArgs/assume-role
=== CONT  TestConfigValuesEqual/equal-multiple-keys
=== CONT  TestConfigValuesEqual/equal-one-key
=== CONT  TestConfigValuesEqual/equal-empty-and-nil-backend-config
=== CONT  TestConfigValuesEqual/equal-empty-and-nil
=== CONT  TestConfigValuesEqual/unequal-values
DEBU[0000] Changed from map[bucket:foo location:europe-west3 prefix:bar project:foo-123456] to map[bucket:foo location:different prefix:bar project:foo-123456] 
=== CONT  TestConfigValuesEqual/unequal-general-bool-handling
=== CONT  TestConfigValuesEqual/unequal-wrong-backend
=== CONT  TestConfigValuesEqual/unequal-non-empty-config-nil
=== CONT  TestConfigValuesEqual/equal-encrypt-bool-handling
=== CONT  TestConfigValuesEqual/equal-ignore-s3-tags
--- PASS: TestForcePathStyleClientSession (0.00s)
    --- PASS: TestForcePathStyleClientSession/path-style-false (0.00s)
    --- PASS: TestForcePathStyleClientSession/path-style-true (0.00s)
    --- PASS: TestForcePathStyleClientSession/path-style-non-existent (0.00s)
--- PASS: TestParseTerraformStateInvalid (0.00s)
--- PASS: TestValidateS3Config (0.00s)
    --- PASS: TestValidateS3Config/no-region (0.00s)
    --- PASS: TestValidateS3Config/no-bucket (0.00s)
    --- PASS: TestValidateS3Config/no-key (0.00s)
--- PASS: TestGCSConfigValuesEqual (0.00s)
    --- PASS: TestGCSConfigValuesEqual/equal-both-empty (0.00s)
    --- PASS: TestGCSConfigValuesEqual/unequal-wrong-backend (0.00s)
    --- PASS: TestGCSConfigValuesEqual/equal-ignore-gcs-labels (0.00s)
    --- PASS: TestGCSConfigValuesEqual/equal-general-bool-handling (0.00s)
    --- PASS: TestGCSConfigValuesEqual/equal-encrypt-bool-handling (0.00s)
    --- PASS: TestGCSConfigValuesEqual/equal-one-key (0.00s)
    --- PASS: TestGCSConfigValuesEqual/equal-multiple-keys (0.00s)
    --- PASS: TestGCSConfigValuesEqual/unequal-non-empty-config-nil (0.00s)
    --- PASS: TestGCSConfigValuesEqual/equal-empty-and-nil (0.00s)
    --- PASS: TestGCSConfigValuesEqual/terragrunt-only-configs-remain-intact (0.00s)
    --- PASS: TestGCSConfigValuesEqual/equal-null-ignored (0.00s)
    --- PASS: TestGCSConfigValuesEqual/unequal-general-bool-handling (0.00s)
    --- PASS: TestGCSConfigValuesEqual/equal-empty-and-nil-backend-config (0.00s)
    --- PASS: TestGCSConfigValuesEqual/unequal-values (0.00s)
DEBU[0000] Backend type has changed from s3 to wrong    
DEBU[0000] Backend config changed from map[foo:different] to map[foo:bar] 
--- PASS: TestDiffersFrom (0.00s)
    --- PASS: TestDiffersFrom/both_empty (0.00s)
    --- PASS: TestDiffersFrom/different_s3_bucket_values (0.00s)
    --- PASS: TestDiffersFrom/different_gcs_bucket_values (0.00s)
    --- PASS: TestDiffersFrom/identical_GCS_configs (0.00s)
    --- PASS: TestDiffersFrom/same_backend_type_value (0.00s)
    --- PASS: TestDiffersFrom/identical_S3_configs (0.00s)
    --- PASS: TestDiffersFrom/gcs_null_values_ignored (0.00s)
    --- PASS: TestDiffersFrom/different_gcs_prefix_values (0.00s)
    --- PASS: TestDiffersFrom/different_s3_key_values (0.00s)
    --- PASS: TestDiffersFrom/different_boolean_values_and_boolean_conversion (0.00s)
    --- PASS: TestDiffersFrom/different_gcs_boolean_values_and_boolean_conversion (0.00s)
    --- PASS: TestDiffersFrom/different_s3_region_values (0.00s)
    --- PASS: TestDiffersFrom/different_backend_type_values (0.00s)
    --- PASS: TestDiffersFrom/null_values_ignored (0.00s)
    --- PASS: TestDiffersFrom/different_gcs_location_values (0.00s)
DEBU[0000] Backend config changed from map[something:%!s(bool=false)] to map[something:%!s(bool=true)] 
--- PASS: TestConfigValuesEqual (0.00s)
    --- PASS: TestConfigValuesEqual/equal-null-ignored (0.00s)
    --- PASS: TestConfigValuesEqual/equal-both-empty (0.00s)
    --- PASS: TestConfigValuesEqual/equal-ignore-dynamodb-tags (0.00s)
    --- PASS: TestConfigValuesEqual/equal-ignore-accesslogging-options (0.00s)
    --- PASS: TestConfigValuesEqual/equal-general-bool-handling (0.00s)
    --- PASS: TestConfigValuesEqual/equal-multiple-keys (0.00s)
    --- PASS: TestConfigValuesEqual/equal-one-key (0.00s)
    --- PASS: TestConfigValuesEqual/equal-empty-and-nil-backend-config (0.00s)
    --- PASS: TestConfigValuesEqual/equal-empty-and-nil (0.00s)
    --- PASS: TestConfigValuesEqual/unequal-non-empty-config-nil (0.00s)
    --- PASS: TestConfigValuesEqual/equal-encrypt-bool-handling (0.00s)
    --- PASS: TestConfigValuesEqual/equal-ignore-s3-tags (0.00s)
    --- PASS: TestConfigValuesEqual/unequal-wrong-backend (0.00s)
    --- PASS: TestConfigValuesEqual/unequal-values (0.00s)
    --- PASS: TestConfigValuesEqual/unequal-general-bool-handling (0.00s)
--- PASS: TestGetTerraformInitArgs (0.00s)
    --- PASS: TestGetTerraformInitArgs/empty-no-values (0.00s)
    --- PASS: TestGetTerraformInitArgs/empty-no-values-all-terragrunt-keys-filtered (0.00s)
    --- PASS: TestGetTerraformInitArgs/terragrunt-keys-filtered (0.00s)
    --- PASS: TestGetTerraformInitArgs/valid-s3-configuration-keys (0.00s)
    --- PASS: TestGetTerraformInitArgs/dynamodb-table-not-replaced-with-lock-table (0.00s)
    --- PASS: TestGetTerraformInitArgs/assume-role (0.00s)
    --- PASS: TestGetTerraformInitArgs/lock-table-replaced-with-dynamodb-table (0.00s)
PASS
ok      github.com/gruntwork-io/terragrunt/remote       0.016s
```

**Review by yhakbar — APPROVED** [review]
Re-tested and it passed. Thank you so much, @findmyname666 !

Great work. I really appreciate the effort and time you put into testing this.

**Pull request merged**